### ISSUE TITLE
I always get message "Very bad performance"

### ISSUE THREAD

**Issue by berarma** [mad3001/Multiply#3]
I have an SD with many files, around 350.

SD  filesystem information:
```
Device:					/dev/sde1
Type:					FAT32
Sector size:				512 bytes
FAT size:				7452 sectors (3815424 bytes)
Number of FATs:				2 
Cluster size:				32768 bytes
Max. cluster chain length:		131071 clusters
Data clusters (total / used / bad):	953824 / 699 / 0
FS size:				29807,00 MiBytes
FAT32 root first cluster:		0x2
First cluster data offset:		0x800000
First cluster FAT entry:		0xfffffff
```

I have tried the command `fatsort -a /dev/sde1` on the SD. It's a Linux command to sort FAT entries.

I still get the same message saying:
```
Folder with way too many hidden,
deleted or incompatible files.
Very bad performance.
```

What can I do?

Many thanks for this tool.

Cheers.

**Comment by mad3001** [comment]
Hi @berarma.
The degradation calculation is done automatically by Multiply when you launch its menu and in each directory you browse.
The calculation for degradation is based on the number of valid files and subdirectories versus the total number of files and subdirectories in that directory (or root directory). Also entries for deleted files and/or subdirectories are counted as tot files, so having a lot of these can also increase de value of degradation.
  Calc = tot files / valid files
The second calculation is done with respect to whether the files are in order on the sdcard. If there are a lot of files that are unsorted (the files were copied without alphabetical order) degradation increase.
  Calc = tot files / sorted files

	total files counts not only files in the folder, but also hidden files.
	valid files counts files with a valid format: SCR, Z80, SNA, TAP. Multiply also checks the validity of these files (based on the information contained in the file).
	If the calc value is greater than 4, the degradation is considered low "degraded performance".
	If the calc value is higher than 8 then the degradation is considered high "very bad performance".
So the first thing is to try to have only valid files and the second thing is to try to have them copied in order within the directory.
To do this it is best to start by copying all the files from the sdcard to your internal hard drive (or ssd) on your computer. Then format the sdcard and then copy the files in order (using a desktop environment cand help to copy files in order).
This should solve this warning.

**Comment by berarma** [comment]
I've done exactly as you said. Also, I've split the files in directories so it's easier to spot issues since there are so many files.

But I still get the same message, except on empty folders or folders with just one file.

I've looked at the code and it seems that the sorting has to be reversed. I tried a directory with reversed order and the message isn't shown for that directory.

It's the condition [here](https://github.com/mad3001/Multiply/blob/60031c23344e7ebee496926ec9b4bb5c5c290071/Src/Multiply_Arduino/libFileList.ino#L60) and [here](https://github.com/mad3001/Multiply/blob/60031c23344e7ebee496926ec9b4bb5c5c290071/Src/Multiply_Arduino/libFileList.ino#L85) doing the reverse sorting check.
